src/signatures/secp256k1.rs

```diff
@@ -0,0 +1,168 @@
+
+        EncodedPoint::from_untagged_bytes(bytes.into())
+            .decode()
+            .map(Self)
+            .map_err(|_| crate::Error::ConvertError {
+                from: "bytes",
+                to: "Secp256k1 Public Key",
+            })
+    }
+
+    /// Load a [PublicKey] from big-endian serialized coordinates.
+    pub fn from_coord(x: &[u8], y: &[u8]) -> crate::Result<Self> {
+        let x: &FieldBytes = x.try_into().map_err(|_| crate::Error::ConvertError {
+            from: "bytes",
+            to: "Secp256k1 Point (x)",
+        })?;
+
+        let y: &FieldBytes = y.try_into().map_err(|_| crate::Error::ConvertError {
+            from: "bytes",
+            to: "Secp256k1 Point (y)",
+        })?;
+
+        EncodedPoint::from_affine_coordinates(x, y, false)
+            .try_into()
+            .map(Self)
+            .map_err(|_| crate::Error::ConvertError {
+                from: "Coordinates",
+                to: "Secp256k1 Public Key",
+            })
+    }
+
+    /// Return the [PublicKey] as an array of bytes.
+    pub fn to_bytes(&self) -> GenericArray<u8, UntaggedPointSize<Secp256k1>> {
+        let point: EncodedPoint<Secp256k1> = self.0.to_encoded_point(false);
+
+        // sanity check encoded point
+        assert!(!point.is_compressed());
+        assert!(!point.is_identity());
+
+        // unwrap is okay - non-identity points are guaranteed by `PublicKey`.
+        point.to_untagged_bytes().unwrap()
+    }
+
+    /// Serialize this public key as SEC1 encoded coordinates.
+    pub fn to_coord(&self) -> crate::Result<(FieldBytes, FieldBytes)> {
+        match self.0.to_encoded_point(false).coordinates() {
+            Coordinates::Uncompressed { x, y } => Ok((*x, *y)),
+            Coordinates::Identity | Coordinates::Compressed { .. } => Err(crate::Error::ConvertError {
+                from: "Secp256k1 Public Key",
+                to: "Coordinates",
+            }),
+        }
+    }
+
+    /// Verify that the given `message` and `signature` are authentic.
+    pub fn verify(&self, message: &[u8], signature: &[u8]) -> crate::Result<()> {
+        signature::Signature::from_bytes(signature)
+            .and_then(|signature| Verifier::verify(self, message, &signature))
+            .map_err(|_| crate::Error::SignatureError { alg: NAME })
+    }
+}
+
+impl Verifier<Signature> for PublicKey {
+    fn verify(&self, message: &[u8], signature: &Signature) -> Result<(), SigError> {
+        k256::ecdsa::VerifyingKey::from(self.0.as_affine()).verify(message, &signature.0)
+    }
+}
+
+/// Secp256k1 Secret Key
+pub struct SecretKey(k256::SecretKey);
+
+impl SecretKey {
+    /// Generate a new random [SecretKey].
+    #[cfg(feature = "random")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "random")))]
+    pub fn generate() -> crate::Result<Self> {
+        let mut bytes: FieldBytes = Default::default();
+
+        crate::utils::rand::fill(&mut bytes[..])?;
```

Comment:
`bytes` will leak secret key bytes. It should be zeroized. This issue is addressed in PR #79.